precommit: enable `end-of-file-fixer`

Borda opened the pull request:
let's formalize suggestion from https://github.com/gitpython-developers/GitPython/pull/1888#discussion_r1615317919

EliahKagan reviewed (changes requested):
Although I think the underlying concept here is good, either all or all but one of the changes inside `test/fixtures` should be omitted.

Most files in `test/fixtures` are test data. Test data should rarely if ever modified for stylistic reasons, because it can break tests, including in ways whose impact would only occur in the case of a regression that a test would otherwise catch. In other words, changing test data without fully and carefully examining exactly how that change might relate to *each* use of the data in the tests can cause tests to break in ways that running the tests does not reveal.

The changes here *do* break the tests. Fortunately, they break them in a way that is easy to catch: a test fails that is supposed to pass. They may or may not *also* break the tests in other ways that are not easy to catch, i.e., that don't affect their status now, but that make the tests pass in slightly *more* circumstances than they should, thereby weakening them against future inadvertent bugs.

In the case of `test/fixtures/.gitconfig`, it may be worthwhile to look into this and maybe make the change anyway, since that file is not just used as a fixture: it is also used to configure CI. In the case of all other files in the `test/fixtures` directory that are modified here, I do not think there is sufficient benefit to justify the added time and effort that it would take to achieve a reasonable level of confidence that the changes would not cause any tests to continue passing in any circumstances under which they should start failing.

Although no `*.py` files inside `test/fixtures` are changed here, those files, while they are considered fixtures, are not test data, so I think it is fine to perform these kinds of stylistic checks and fixes on them. Other files in `test/fixtures`, with the *possible* exception of `test/fixtures/.gitconfig`, really should not be subject to these kinds of checks or receive these kinds of changes.

See also https://github.com/gitpython-developers/GitPython/pull/1888#discussion_r1615317787.

The changes outside of `test/fixtures`, including the automated changes in `doc/source`, all look good to me.

Byron reviewed (changes requested):
Thanks for the initiative!

I am now officially setting this PR to a state that indicates that some modifications are needed.

Borda commented:
> The changes outside of `test/fixtures`, including the automated changes in `doc/source`, all look good to me.

reverted the changes in `tests/fixtures` and added this folder to be ignored

EliahKagan reviewed (approved):
LGTM

Borda commented:
@Byron mind have a look again pls

Byron reviewed (approved):
Thanks a lot!
